yxi/axi-calyx/axi-combined-calyx.futil

```diff
@@ -446,21 +443,21 @@ component m_awwrite_channel(
       // this contains blocking logic previously in its own group
       group do_aw_transfer {
           //assert AWVALID
-          is_awvalid.in = !awvalid_was_high.out ? 1'b1;
+          is_awvalid.in = !(is_awvalid.out & AWREADY) & !aw_handshake_occurred.out ? 1'b1;
           
-          // TODO(nathanielnrn): in theory should be able to get rid of awvalid_was_high
+          // TODO(nathanielnrn): in theory should be able to get rid of aw_handshake_occurred
           // but for now we will be explicit and reduce this in generation maybe. Not sure
           // it even matters.
           // This makes AWVALID go low after a single cycle. Without it it stays high for 2.
-          is_awvalid.in = is_awvalid.out & AWREADY & awvalid_was_high.out ? 1'b0;
+          is_awvalid.in = (is_awvalid.out & AWREADY) | aw_handshake_occurred.out ? 1'b0;
```

Comment:
The condition seems like the "De Morgan's negative" of the other condition for `is_awvalid.in`. Perhaps, in the generator, we want a utility to emit this pattern:

```
lhs = cond ? 1;
lhs = !cond ? 0;
```

…with a single call to, like, `assign_logic(lhs, cond)` or something like that. Could make things a little more readable at the generator level.

Reply:
Maybe something like `partition_assignment(lhs, cond)`?